server/src/sim.rs

```diff
@@ -185,6 +190,7 @@ impl Sim {
                 {
                     current_node = self.graph.ensure_neighbor(current_node, side);
                 }
+                self.populate_fresh_graph_nodes();
```

Comment:
Would it be tidier to do this once at the end of `load_all_entities` rather than separately for every character?

Reply:
My main concern is that `load_component` puts `server::Sim` in a weird state, and I expect it to be easier to reason about the code-long term if the reason for calling `populate_fresh_graph_nodes` is next to where it's called.

However, I goofed up, as `populate_fresh_graph_nodes` is not designed to be called multiple times in a single `step`. I believe that is the part that I ought to fix.